### ISSUE TITLE
iOS a11y gets stuck in nested scrollables

### ISSUE THREAD

**Issue by dnfield** [flutter/flutter#92853]
Reproduction:

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({Key? key}) : super(key: key);

  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: const MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  const MyHomePage({Key? key, required this.title}) : super(key: key);

  final String title;

  @override
  State<MyHomePage> createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text(widget.title),
      ),
      body: Center(
        child: convoListView(),
      ),
    );
  }
}

Widget convoListView() {
  return ListView(
    children: [
      renderProfileCard(),
      Container(
        height: 400.0,
        color: Colors.cyan,
        child: Text('fake card'),
      ),
      Container(
        height: 400.0,
        color: Colors.green,
        child: Text('fake card'),
      ),
      Container(
        height: 400.0,
        color: Colors.brown,
        child: Text('fake card'),
      ),
      Container(
        height: 400.0,
        color: Colors.cyan,
        child: Text('fake card'),
      ),
    ],
  );
}

Widget renderProfileCard() {
  // Merges semantics for profile card, so accessibility focus is correct
  // even after scrolling.
  return MergeSemantics(
    child: Builder(
      builder: (context) => defaultEmptyStateWidget(),
    ),
  );
}

Widget defaultEmptyStateWidget() {
  return SingleChildScrollView(
    child: Column(
      mainAxisAlignment: MainAxisAlignment.center,
      children: [
        ExcludeSemantics(
          child: Container(
            height: 300.0,
            color: Colors.blueGrey,
            child: Text('fake avatar'),
          ),
        ),
        Container(
          height: 100.0,
          color: Colors.grey,
          child: Text('fake name'),
        ),
      ],
    ),
  );
}
```

Produces logs like: 

> [Accessibility] Went all the way up the container chain from @, but could not find any container that was one of the ordered children of @.  This may be acceptable if it happened right around a layout change, but it would be best to double check by swiping left/right to see if you can get to all elements.

When it gets stuck

/cc @chunhtai @jmagman

**Comment by dnfield** [comment]
This also creates a crash in some cases where I turn a11y off.

**Comment by dnfield** [comment]
xref b/200870816

**Comment by dnfield** [comment]
Internal customer was able to work around this by not using the nested scroll view. Since this is no longer blocking them, I'm going to deprioritize. We should definitely fix this, but when @chunhtai is back I think he'll have more context than I do.

**Comment by dnfield** [comment]
Assigning this to chuntai so we don't forget to talk about it.

**Comment by dnfield** [comment]
This also reproduces with a WebView:

```dart
import 'package:flutter/material.dart';
import 'package:webview_flutter/webview_flutter.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      home: Scaffold(
        appBar: AppBar(
          title: const Text('Demo'),
        ),
        body: WebView(
          initialUrl: 'https://google.com',
          javascriptMode: JavascriptMode.disabled,
        ),
      ),
    );
  }
}
```

Unfortunately, this one is not as easy to work around.

**Comment by jmagman** [comment]
Bisected the `WebView` repro case to https://github.com/flutter/engine/pull/24308.  We need to make sure we don't regress https://github.com/flutter/flutter/issues/73716.

**Comment by cyanglaz** [comment]
> Bisected the `WebView` repro case to [flutter/engine#24308](https://github.com/flutter/engine/pull/24308). We need to make sure we don't regress #73716.

They probably not related. flutter/engine#24308 only updated the `FlutterPlatformViewSemanticsContainer`, which only exists in PlatformView scenarios.

**Comment by dnfield** [comment]
@cyanglaz the WebView case is using a PlatformView.

**Comment by cyanglaz** [comment]
> @cyanglaz the WebView case is using a PlatformView.

Yes, i meant the webview issue in  https://github.com/flutter/flutter/issues/92853#issuecomment-959719752 and the original issue might not be related.

**Comment by jmagman** [comment]
I'll file a new issue for just the PlatformView repro case.

**Comment by dnfield** [comment]
SG. It's likely that these both are using the same incorrect pattern though

**Comment by cyanglaz** [comment]
> SG. It's likely that these both are using the same incorrect pattern though

Yeah, once i found the solution to #93024, I can also check if the solution can help with this.

**Comment by jmagman** [comment]
> Assigning this to chuntai so we don't forget to talk about it.


and back to a P4 since the actual blocker is https://github.com/flutter/flutter/issues/93024.

**Comment by cyanglaz** [comment]
@dnfield Moving the [discussion](https://github.com/flutter/engine/pull/29531/files#issuecomment-963550048) here: 

> @cyanglaz can any of this be applied to the FlutterSemanticsScrollView implementation? It seems like there may be som eoverlap but I feel like I'm missing a piece.

They seem to be unrelated. One thing we can do for the scroll view issue is to figure out the crash and see if it it related to the accessibility. (Sometimes crashes are easier to be analyzed than accessibility :) )

Also, I saw that we are declaring the children count for scroll view [here](https://github.com/flutter/engine/blob/master/shell/platform/darwin/ios/framework/Source/FlutterSemanticsScrollView.mm#L101), but not each children (e.g. accessibilityElementAtIndex: not implemented). I'm not sure what does this pattern do.
